diff --git a/packages/flutter/lib/src/widgets/navigator.dart b/packages/flutter/lib/src/widgets/navigator.dart
index c2e583e849c..658b773d096 100644
--- a/packages/flutter/lib/src/widgets/navigator.dart
+++ b/packages/flutter/lib/src/widgets/navigator.dart
@@ -3308,6 +3308,7 @@ class _RouteEntry extends RouteTransitionRecord {
           .then((dynamic result) async {
             switch (defaultTargetPlatform) {
               case TargetPlatform.android:
+              case TargetPlatform.ohos:
                 // In the Android platform, we have to wait for the system refocus to complete before
                 // sending the refocus message. Otherwise, the refocus message will be ignored.
                 // TODO(hangyujin): update this logic if Android provide a better way to do so.
diff --git a/packages/flutter/lib/src/widgets/scrollbar.dart b/packages/flutter/lib/src/widgets/scrollbar.dart
index 12d3aa444df..c85e5abdf94 100644
--- a/packages/flutter/lib/src/widgets/scrollbar.dart
+++ b/packages/flutter/lib/src/widgets/scrollbar.dart
@@ -1890,7 +1890,7 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
     // dragging on the trackpad or with a stylus.
     final TargetPlatform platform = ScrollConfiguration.of(context).getPlatform(context);
     final Velocity adjustedVelocity = switch (platform) {
-      TargetPlatform.iOS || TargetPlatform.android => -velocity,
+      TargetPlatform.iOS || TargetPlatform.android || TargetPlatform.ohos => -velocity,
       _ => Velocity.zero,
     };
     final renderBox = _scrollbarPainterKey.currentContext!.findRenderObject()! as RenderBox;
