diff --git a/packages/flutter/lib/src/cupertino/adaptive_text_selection_toolbar.dart b/packages/flutter/lib/src/cupertino/adaptive_text_selection_toolbar.dart
index e42d42d6f8a..7d628bfcf0d 100644
--- a/packages/flutter/lib/src/cupertino/adaptive_text_selection_toolbar.dart
+++ b/packages/flutter/lib/src/cupertino/adaptive_text_selection_toolbar.dart
@@ -192,6 +192,7 @@ class CupertinoAdaptiveTextSelectionToolbar extends StatelessWidget {
   ) {
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.iOS:
       case TargetPlatform.ohos:
@@ -219,6 +220,7 @@ class CupertinoAdaptiveTextSelectionToolbar extends StatelessWidget {
 
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.iOS:
       case TargetPlatform.fuchsia:
       case TargetPlatform.ohos:
diff --git a/packages/flutter/lib/src/cupertino/app.dart b/packages/flutter/lib/src/cupertino/app.dart
index 3941bc1e0d7..53362301eda 100644
--- a/packages/flutter/lib/src/cupertino/app.dart
+++ b/packages/flutter/lib/src/cupertino/app.dart
@@ -476,6 +476,7 @@ class CupertinoScrollBehavior extends ScrollBehavior {
         return CupertinoScrollbar(controller: details.controller, child: child);
       case TargetPlatform.ohos:
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.iOS:
         return child;
diff --git a/packages/flutter/lib/src/cupertino/button.dart b/packages/flutter/lib/src/cupertino/button.dart
index da92f85bcc5..7fdc8486851 100644
--- a/packages/flutter/lib/src/cupertino/button.dart
+++ b/packages/flutter/lib/src/cupertino/button.dart
@@ -301,7 +301,7 @@ class CupertinoButton extends StatefulWidget {
   static double tapMoveSlop() {
     return switch (defaultTargetPlatform) {
       TargetPlatform.iOS ||
-      TargetPlatform.android ||
+      TargetPlatform.android || TargetPlatform.ohos ||
       TargetPlatform.fuchsia ||
       TargetPlatform.ohos => kCupertinoButtonTapMoveSlop,
       TargetPlatform.macOS || TargetPlatform.linux || TargetPlatform.windows => 0.0,
diff --git a/packages/flutter/lib/src/cupertino/checkbox.dart b/packages/flutter/lib/src/cupertino/checkbox.dart
index 8527ce84275..c1ab38377fd 100644
--- a/packages/flutter/lib/src/cupertino/checkbox.dart
+++ b/packages/flutter/lib/src/cupertino/checkbox.dart
@@ -442,7 +442,7 @@ class _CupertinoCheckboxState extends State<CupertinoCheckbox>
         switch (defaultTargetPlatform) {
           TargetPlatform.iOS ||
           TargetPlatform.ohos ||
-          TargetPlatform.android ||
+          TargetPlatform.android || TargetPlatform.ohos ||
           TargetPlatform.fuchsia => const Size.square(kMinInteractiveDimensionCupertino),
           TargetPlatform.macOS ||
           TargetPlatform.linux ||
diff --git a/packages/flutter/lib/src/cupertino/expansion_tile.dart b/packages/flutter/lib/src/cupertino/expansion_tile.dart
index 33eee192319..877c33a9139 100644
--- a/packages/flutter/lib/src/cupertino/expansion_tile.dart
+++ b/packages/flutter/lib/src/cupertino/expansion_tile.dart
@@ -180,6 +180,7 @@ class _CupertinoExpansionTileState extends State<CupertinoExpansionTile> {
             ? '${localizations.collapsedHint}\n ${localizations.expansionTileExpandedHint}'
             : '${localizations.expandedHint}\n ${localizations.expansionTileCollapsedHint}';
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/cupertino/menu_anchor.dart b/packages/flutter/lib/src/cupertino/menu_anchor.dart
index 76fae397c27..6ae027988a6 100644
--- a/packages/flutter/lib/src/cupertino/menu_anchor.dart
+++ b/packages/flutter/lib/src/cupertino/menu_anchor.dart
@@ -35,6 +35,7 @@ bool get _isCupertino {
     case TargetPlatform.macOS:
       return true;
     case TargetPlatform.android:
+        case TargetPlatform.ohos:
     case TargetPlatform.fuchsia:
     case TargetPlatform.linux:
     case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/cupertino/picker.dart b/packages/flutter/lib/src/cupertino/picker.dart
index 8e7d7caba87..9c398e1a9c6 100644
--- a/packages/flutter/lib/src/cupertino/picker.dart
+++ b/packages/flutter/lib/src/cupertino/picker.dart
@@ -277,6 +277,7 @@ class _CupertinoPickerState extends State<CupertinoPicker> {
           SystemSound.play(SystemSoundType.tick);
         }
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.macOS:
diff --git a/packages/flutter/lib/src/cupertino/slider.dart b/packages/flutter/lib/src/cupertino/slider.dart
index 445ccfc1ffb..3553d7359df 100644
--- a/packages/flutter/lib/src/cupertino/slider.dart
+++ b/packages/flutter/lib/src/cupertino/slider.dart
@@ -266,6 +266,7 @@ class _CupertinoSliderState extends State<CupertinoSlider> with TickerProviderSt
           HapticFeedback.selectionClick();
         }
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.macOS:
diff --git a/packages/flutter/lib/src/cupertino/switch.dart b/packages/flutter/lib/src/cupertino/switch.dart
index 24b1ed97f27..3072d92cf61 100644
--- a/packages/flutter/lib/src/cupertino/switch.dart
+++ b/packages/flutter/lib/src/cupertino/switch.dart
@@ -660,6 +660,7 @@ class _CupertinoSwitchState extends State<CupertinoSwitch>
       case TargetPlatform.iOS:
         HapticFeedback.lightImpact();
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.macOS:
diff --git a/packages/flutter/lib/src/cupertino/text_field.dart b/packages/flutter/lib/src/cupertino/text_field.dart
index 6bc575bf690..bfbe6e7872b 100644
--- a/packages/flutter/lib/src/cupertino/text_field.dart
+++ b/packages/flutter/lib/src/cupertino/text_field.dart
@@ -1056,6 +1056,7 @@ class CupertinoTextField extends StatefulWidget {
         ) {
           switch (defaultTargetPlatform) {
             case TargetPlatform.android:
+        case TargetPlatform.ohos:
             case TargetPlatform.iOS:
             case TargetPlatform.ohos:
               return CupertinoTextMagnifier(controller: controller, magnifierInfo: magnifierInfo);
@@ -1461,6 +1462,7 @@ class _CupertinoTextFieldState extends State<CupertinoTextField>
     switch (defaultTargetPlatform) {
       case TargetPlatform.iOS:
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.ohos:
         textSelectionControls ??= cupertinoTextSelectionHandleControls;
diff --git a/packages/flutter/lib/src/material/about.dart b/packages/flutter/lib/src/material/about.dart
index 51f29bae395..d5b6c5e468d 100644
--- a/packages/flutter/lib/src/material/about.dart
+++ b/packages/flutter/lib/src/material/about.dart
@@ -522,6 +522,7 @@ class _AdaptiveAboutDialog extends AboutDialog {
           ),
         ];
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/material/action_buttons.dart b/packages/flutter/lib/src/material/action_buttons.dart
index 7f4c4d3d50f..d62cb05bfc6 100644
--- a/packages/flutter/lib/src/material/action_buttons.dart
+++ b/packages/flutter/lib/src/material/action_buttons.dart
@@ -93,6 +93,7 @@ class _ActionIcon extends StatelessWidget {
     // expects the duplicated tooltip and label.
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
         semanticsLabel = getAndroidSemanticsLabel(MaterialLocalizations.of(context));
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
@@ -137,6 +138,7 @@ class BackButtonIcon extends StatelessWidget {
         }
         switch (Theme.of(context).platform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.linux:
           case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/material/adaptive_text_selection_toolbar.dart b/packages/flutter/lib/src/material/adaptive_text_selection_toolbar.dart
index 7e7acf90cc7..83f06a04c22 100644
--- a/packages/flutter/lib/src/material/adaptive_text_selection_toolbar.dart
+++ b/packages/flutter/lib/src/material/adaptive_text_selection_toolbar.dart
@@ -212,6 +212,7 @@ class AdaptiveTextSelectionToolbar extends StatelessWidget {
       case TargetPlatform.macOS:
         return CupertinoTextSelectionToolbarButton.getButtonLabel(context, buttonItem);
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.ohos:
@@ -318,6 +319,7 @@ class AdaptiveTextSelectionToolbar extends StatelessWidget {
         );
       case TargetPlatform.ohos:
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
         return TextSelectionToolbar(
           anchorAbove: anchors.primaryAnchor,
           anchorBelow: anchors.secondaryAnchor == null
diff --git a/packages/flutter/lib/src/material/app.dart b/packages/flutter/lib/src/material/app.dart
index dfdf9b0b419..67a6e9cf6a0 100644
--- a/packages/flutter/lib/src/material/app.dart
+++ b/packages/flutter/lib/src/material/app.dart
@@ -868,6 +868,7 @@ class MaterialScrollBehavior extends ScrollBehavior {
             assert(details.controller != null);
             return Scrollbar(controller: details.controller, child: child);
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.iOS:
           case TargetPlatform.ohos:
diff --git a/packages/flutter/lib/src/material/app_bar.dart b/packages/flutter/lib/src/material/app_bar.dart
index 5dadad27046..f4d4f04ea71 100644
--- a/packages/flutter/lib/src/material/app_bar.dart
+++ b/packages/flutter/lib/src/material/app_bar.dart
@@ -806,7 +806,7 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
     bool platformCenter() {
       return switch (theme.platform) {
         TargetPlatform.iOS || TargetPlatform.macOS => actions == null || actions!.length < 2,
-        TargetPlatform.android ||
+        TargetPlatform.android || TargetPlatform.ohos ||
         TargetPlatform.fuchsia ||
         TargetPlatform.linux ||
         TargetPlatform.ohos ||
@@ -1071,7 +1071,7 @@ class _AppBarState extends State<AppBar> {
       if (!widget.excludeHeaderSemantics) {
         title = Semantics(
           namesRoute: switch (defaultTargetPlatform) {
-            TargetPlatform.android ||
+            TargetPlatform.android || TargetPlatform.ohos ||
             TargetPlatform.fuchsia ||
             TargetPlatform.linux ||
             TargetPlatform.windows ||
diff --git a/packages/flutter/lib/src/material/bottom_sheet.dart b/packages/flutter/lib/src/material/bottom_sheet.dart
index 45a077e8f56..969b686167f 100644
--- a/packages/flutter/lib/src/material/bottom_sheet.dart
+++ b/packages/flutter/lib/src/material/bottom_sheet.dart
@@ -721,7 +721,7 @@ class _ModalBottomSheetState<T> extends State<_ModalBottomSheet<T>> {
 
   String _getRouteLabel(MaterialLocalizations localizations) => switch (defaultTargetPlatform) {
     TargetPlatform.iOS || TargetPlatform.macOS => '',
-    TargetPlatform.android ||
+    TargetPlatform.android || TargetPlatform.ohos ||
     TargetPlatform.fuchsia ||
     TargetPlatform.linux ||
     TargetPlatform.windows ||
diff --git a/packages/flutter/lib/src/material/calendar_date_picker.dart b/packages/flutter/lib/src/material/calendar_date_picker.dart
index a32123796b9..f13ba1a5cc0 100644
--- a/packages/flutter/lib/src/material/calendar_date_picker.dart
+++ b/packages/flutter/lib/src/material/calendar_date_picker.dart
@@ -259,6 +259,7 @@ class _CalendarDatePickerState extends State<CalendarDatePicker> {
   void _vibrate() {
     switch (Theme.of(context).platform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
@@ -335,6 +336,7 @@ class _CalendarDatePickerState extends State<CalendarDatePicker> {
             _textDirection,
           ).catchError(_reportAnnouncementError);
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.iOS:
         case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
diff --git a/packages/flutter/lib/src/material/checkbox.dart b/packages/flutter/lib/src/material/checkbox.dart
index 2ebccabd9a8..dea4940eeaa 100644
--- a/packages/flutter/lib/src/material/checkbox.dart
+++ b/packages/flutter/lib/src/material/checkbox.dart
@@ -479,6 +479,7 @@ class _CheckboxState extends State<Checkbox> with TickerProviderStateMixin, Togg
         final ThemeData theme = Theme.of(context);
         switch (theme.platform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.linux:
           case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/material/date_picker.dart b/packages/flutter/lib/src/material/date_picker.dart
index 34f3edab02d..1ccda95c72c 100644
--- a/packages/flutter/lib/src/material/date_picker.dart
+++ b/packages/flutter/lib/src/material/date_picker.dart
@@ -2065,6 +2065,7 @@ class _CalendarDateRangePickerState extends State<_CalendarDateRangePicker> {
   void _vibrate() {
     switch (Theme.of(context).platform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
         HapticFeedback.vibrate();
       case TargetPlatform.iOS:
diff --git a/packages/flutter/lib/src/material/dialog.dart b/packages/flutter/lib/src/material/dialog.dart
index 579f82c0a61..dd139f9114c 100644
--- a/packages/flutter/lib/src/material/dialog.dart
+++ b/packages/flutter/lib/src/material/dialog.dart
@@ -773,7 +773,7 @@ class AlertDialog extends StatelessWidget {
 
     final String? label = switch (defaultTargetPlatform) {
       TargetPlatform.iOS || TargetPlatform.macOS => semanticLabel,
-      TargetPlatform.android ||
+      TargetPlatform.android || TargetPlatform.ohos ||
       TargetPlatform.fuchsia ||
       TargetPlatform.linux ||
       TargetPlatform.ohos ||
@@ -1006,6 +1006,7 @@ class _AdaptiveAlertDialog extends AlertDialog {
     final ThemeData theme = Theme.of(context);
     switch (theme.platform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
@@ -1293,6 +1294,7 @@ class SimpleDialog extends StatelessWidget {
       case TargetPlatform.iOS:
         break;
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
@@ -1727,6 +1729,7 @@ Future<T?> showAdaptiveDialog<T>({
   final ThemeData theme = Theme.of(context);
   switch (theme.platform) {
     case TargetPlatform.android:
+        case TargetPlatform.ohos:
     case TargetPlatform.fuchsia:
     case TargetPlatform.linux:
     case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/material/drawer.dart b/packages/flutter/lib/src/material/drawer.dart
index fbbb163dfcc..7b484ccb362 100644
--- a/packages/flutter/lib/src/material/drawer.dart
+++ b/packages/flutter/lib/src/material/drawer.dart
@@ -252,7 +252,7 @@ class Drawer extends StatelessWidget {
     final DrawerThemeData drawerTheme = DrawerTheme.of(context);
     final String? label = switch (defaultTargetPlatform) {
       TargetPlatform.iOS || TargetPlatform.macOS => semanticLabel,
-      TargetPlatform.android ||
+      TargetPlatform.android || TargetPlatform.ohos ||
       TargetPlatform.fuchsia ||
       TargetPlatform.linux ||
       TargetPlatform.ohos ||
@@ -647,7 +647,7 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
 
   Widget _buildDrawer(BuildContext context) {
     final bool isDesktop = switch (Theme.of(context).platform) {
-      TargetPlatform.android ||
+      TargetPlatform.android || TargetPlatform.ohos ||
       TargetPlatform.iOS ||
       TargetPlatform.fuchsia ||
       TargetPlatform.ohos => false,
@@ -686,7 +686,7 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
       }
     } else {
       final bool platformHasBackButton = switch (defaultTargetPlatform) {
-        TargetPlatform.android => true,
+        TargetPlatform.android || TargetPlatform.ohos => true,
         TargetPlatform.iOS ||
         TargetPlatform.ohos ||
         TargetPlatform.macOS ||
diff --git a/packages/flutter/lib/src/material/dropdown_menu.dart b/packages/flutter/lib/src/material/dropdown_menu.dart
index 779ad328b9b..dba1c98c213 100644
--- a/packages/flutter/lib/src/material/dropdown_menu.dart
+++ b/packages/flutter/lib/src/material/dropdown_menu.dart
@@ -816,7 +816,7 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
         widget.requestFocusOnTap ??
         switch (Theme.of(context).platform) {
           TargetPlatform.iOS ||
-          TargetPlatform.android ||
+          TargetPlatform.android || TargetPlatform.ohos ||
           TargetPlatform.fuchsia ||
           TargetPlatform.ohos => false,
           TargetPlatform.macOS || TargetPlatform.linux || TargetPlatform.windows => true,
diff --git a/packages/flutter/lib/src/material/flexible_space_bar.dart b/packages/flutter/lib/src/material/flexible_space_bar.dart
index 12963190bea..86b5575ee8b 100644
--- a/packages/flutter/lib/src/material/flexible_space_bar.dart
+++ b/packages/flutter/lib/src/material/flexible_space_bar.dart
@@ -191,7 +191,7 @@ class _FlexibleSpaceBarState extends State<FlexibleSpaceBar> {
   bool _getEffectiveCenterTitle(ThemeData theme) {
     return widget.centerTitle ??
         switch (theme.platform) {
-          TargetPlatform.android ||
+          TargetPlatform.android || TargetPlatform.ohos ||
           TargetPlatform.fuchsia ||
           TargetPlatform.linux ||
           TargetPlatform.windows ||
@@ -300,6 +300,7 @@ class _FlexibleSpaceBarState extends State<FlexibleSpaceBar> {
             case TargetPlatform.macOS:
               title = widget.title;
             case TargetPlatform.android:
+        case TargetPlatform.ohos:
             case TargetPlatform.fuchsia:
             case TargetPlatform.linux:
             case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/material/icons.dart b/packages/flutter/lib/src/material/icons.dart
index d71c87c0866..6e6856590de 100644
--- a/packages/flutter/lib/src/material/icons.dart
+++ b/packages/flutter/lib/src/material/icons.dart
@@ -20,6 +20,7 @@ final class PlatformAdaptiveIcons implements Icons {
   static bool _isCupertino() {
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/material/menu_anchor.dart b/packages/flutter/lib/src/material/menu_anchor.dart
index e76e6e570ff..4092a2ae18a 100644
--- a/packages/flutter/lib/src/material/menu_anchor.dart
+++ b/packages/flutter/lib/src/material/menu_anchor.dart
@@ -2665,6 +2665,7 @@ class _LocalizedShortcutLabeler {
         modifier == LogicalKeyboardKey.metaRight) {
       switch (defaultTargetPlatform) {
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.ohos:
         case TargetPlatform.linux:
@@ -2681,6 +2682,7 @@ class _LocalizedShortcutLabeler {
         modifier == LogicalKeyboardKey.altRight) {
       switch (defaultTargetPlatform) {
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.linux:
         case TargetPlatform.windows:
@@ -2701,6 +2703,7 @@ class _LocalizedShortcutLabeler {
       // we just return the local translation of "Ctrl".
       switch (defaultTargetPlatform) {
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.linux:
         case TargetPlatform.windows:
@@ -2716,6 +2719,7 @@ class _LocalizedShortcutLabeler {
         modifier == LogicalKeyboardKey.shiftRight) {
       switch (defaultTargetPlatform) {
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.linux:
         case TargetPlatform.windows:
@@ -3987,6 +3991,7 @@ bool get _isCupertino {
     case TargetPlatform.macOS:
       return true;
     case TargetPlatform.android:
+        case TargetPlatform.ohos:
     case TargetPlatform.fuchsia:
     case TargetPlatform.linux:
     case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/material/page.dart b/packages/flutter/lib/src/material/page.dart
index 0880e20b52b..db4ad582c74 100644
--- a/packages/flutter/lib/src/material/page.dart
+++ b/packages/flutter/lib/src/material/page.dart
@@ -103,7 +103,7 @@ mixin MaterialRouteTransitionMixin<T> on PageRoute<T> {
     return pageTransitionsTheme.builders[platform] ??
         switch (platform) {
           TargetPlatform.iOS || TargetPlatform.macOS => const CupertinoPageTransitionsBuilder(),
-          TargetPlatform.android ||
+          TargetPlatform.android || TargetPlatform.ohos ||
           TargetPlatform.fuchsia ||
           TargetPlatform.windows ||
           TargetPlatform.linux ||
diff --git a/packages/flutter/lib/src/material/page_transitions_theme.dart b/packages/flutter/lib/src/material/page_transitions_theme.dart
index 4e0eb78b0b0..54b29f121ae 100644
--- a/packages/flutter/lib/src/material/page_transitions_theme.dart
+++ b/packages/flutter/lib/src/material/page_transitions_theme.dart
@@ -979,7 +979,7 @@ class _PageTransitionsThemeTransitionsState<T> extends State<_PageTransitionsThe
         widget.builders[platform] ??
         switch (platform) {
           TargetPlatform.iOS => const CupertinoPageTransitionsBuilder(),
-          TargetPlatform.android ||
+          TargetPlatform.android || TargetPlatform.ohos ||
           TargetPlatform.fuchsia ||
           TargetPlatform.windows ||
           TargetPlatform.macOS ||
diff --git a/packages/flutter/lib/src/material/radio.dart b/packages/flutter/lib/src/material/radio.dart
index 40fdc77eec6..f264ac67702 100644
--- a/packages/flutter/lib/src/material/radio.dart
+++ b/packages/flutter/lib/src/material/radio.dart
@@ -517,6 +517,7 @@ class _RadioState<T> extends State<Radio<T>> {
         final ThemeData theme = Theme.of(context);
         switch (theme.platform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.linux:
           case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/material/range_slider.dart b/packages/flutter/lib/src/material/range_slider.dart
index 699455b1596..f1fe3bc485c 100644
--- a/packages/flutter/lib/src/material/range_slider.dart
+++ b/packages/flutter/lib/src/material/range_slider.dart
@@ -1228,6 +1228,7 @@ class _RenderRangeSlider extends RenderBox with RelayoutWhenSystemFontsChangeMix
         // Matches iOS implementation of material slider.
         return 0.1;
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.macOS:
diff --git a/packages/flutter/lib/src/material/refresh_indicator.dart b/packages/flutter/lib/src/material/refresh_indicator.dart
index 18e632cb26f..bac75e90a37 100644
--- a/packages/flutter/lib/src/material/refresh_indicator.dart
+++ b/packages/flutter/lib/src/material/refresh_indicator.dart
@@ -684,6 +684,7 @@ class RefreshIndicatorState extends State<RefreshIndicator>
                             final ThemeData theme = Theme.of(context);
                             switch (theme.platform) {
                               case TargetPlatform.android:
+        case TargetPlatform.ohos:
                               case TargetPlatform.fuchsia:
                               case TargetPlatform.linux:
                               case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/material/reorderable_list.dart b/packages/flutter/lib/src/material/reorderable_list.dart
index e2a0fa52908..154c78d9caf 100644
--- a/packages/flutter/lib/src/material/reorderable_list.dart
+++ b/packages/flutter/lib/src/material/reorderable_list.dart
@@ -438,6 +438,7 @@ class _ReorderableListViewState extends State<ReorderableListView> {
 
         case TargetPlatform.iOS:
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.ohos:
           return ReorderableDelayedDragStartListener(key: itemGlobalKey, index: index, child: item);
diff --git a/packages/flutter/lib/src/material/scaffold.dart b/packages/flutter/lib/src/material/scaffold.dart
index ff0cf555115..b6293e972ce 100644
--- a/packages/flutter/lib/src/material/scaffold.dart
+++ b/packages/flutter/lib/src/material/scaffold.dart
@@ -3193,7 +3193,7 @@ class ScaffoldState extends State<Scaffold>
       TargetPlatform.iOS ||
       TargetPlatform.macOS ||
       TargetPlatform.ohos => widget.primary ? _HitTestableAtOrigin(_statusBarKey) : null,
-      TargetPlatform.android ||
+      TargetPlatform.android || TargetPlatform.ohos ||
       TargetPlatform.fuchsia ||
       TargetPlatform.linux ||
       TargetPlatform.windows => null,
diff --git a/packages/flutter/lib/src/material/scrollbar.dart b/packages/flutter/lib/src/material/scrollbar.dart
index 4db807c2804..60372546dc1 100644
--- a/packages/flutter/lib/src/material/scrollbar.dart
+++ b/packages/flutter/lib/src/material/scrollbar.dart
@@ -330,6 +330,7 @@ class _MaterialScrollbarState extends RawScrollbarState<_MaterialScrollbar> {
     _scrollbarTheme = ScrollbarTheme.of(context);
     switch (theme.platform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
         _useAndroidScrollbar = true;
       case TargetPlatform.ohos:
       case TargetPlatform.iOS:
diff --git a/packages/flutter/lib/src/material/search.dart b/packages/flutter/lib/src/material/search.dart
index cdf3f8dcf79..1ead1d8e38d 100644
--- a/packages/flutter/lib/src/material/search.dart
+++ b/packages/flutter/lib/src/material/search.dart
@@ -617,6 +617,7 @@ class _SearchPageState<T> extends State<_SearchPage<T>> {
       case TargetPlatform.macOS:
         routeName = '';
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/material/search_anchor.dart b/packages/flutter/lib/src/material/search_anchor.dart
index 9442633f5f7..a331b752d72 100644
--- a/packages/flutter/lib/src/material/search_anchor.dart
+++ b/packages/flutter/lib/src/material/search_anchor.dart
@@ -555,7 +555,7 @@ class _SearchAnchorState extends State<SearchAnchor> {
     return widget.isFullScreen ??
         switch (Theme.of(context).platform) {
           TargetPlatform.iOS ||
-          TargetPlatform.android ||
+          TargetPlatform.android || TargetPlatform.ohos ||
           TargetPlatform.fuchsia ||
           TargetPlatform.ohos => true,
           TargetPlatform.macOS || TargetPlatform.linux || TargetPlatform.windows => false,
diff --git a/packages/flutter/lib/src/material/selectable_text.dart b/packages/flutter/lib/src/material/selectable_text.dart
index 8a3ece013e4..b2a2ee83095 100644
--- a/packages/flutter/lib/src/material/selectable_text.dart
+++ b/packages/flutter/lib/src/material/selectable_text.dart
@@ -626,6 +626,7 @@ class _SelectableTextState extends State<SelectableText>
         }
         return;
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/material/selection_area.dart b/packages/flutter/lib/src/material/selection_area.dart
index e1412c1f9a5..d315cb64cbe 100644
--- a/packages/flutter/lib/src/material/selection_area.dart
+++ b/packages/flutter/lib/src/material/selection_area.dart
@@ -126,7 +126,7 @@ class SelectionAreaState extends State<SelectionArea> {
     final TextSelectionControls controls =
         widget.selectionControls ??
         switch (Theme.of(context).platform) {
-          TargetPlatform.android || TargetPlatform.fuchsia || TargetPlatform.ohos => materialTextSelectionHandleControls,
+          TargetPlatform.android || TargetPlatform.ohos || TargetPlatform.fuchsia || TargetPlatform.ohos => materialTextSelectionHandleControls,
           TargetPlatform.linux || TargetPlatform.windows => desktopTextSelectionHandleControls,
           TargetPlatform.iOS => cupertinoTextSelectionHandleControls,
           TargetPlatform.macOS => cupertinoDesktopTextSelectionHandleControls,
diff --git a/packages/flutter/lib/src/material/slider.dart b/packages/flutter/lib/src/material/slider.dart
index 73d20f18897..9e4b72f4624 100644
--- a/packages/flutter/lib/src/material/slider.dart
+++ b/packages/flutter/lib/src/material/slider.dart
@@ -946,6 +946,7 @@ class _SliderState extends State<Slider> with TickerProviderStateMixin {
     VoidCallback? handleDidGainAccessibilityFocus;
     switch (theme.platform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.iOS:
       case TargetPlatform.linux:
@@ -1511,6 +1512,7 @@ class _RenderSlider extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
         // Matches iOS implementation of material slider.
         return 0.1;
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.ohos:
       case TargetPlatform.linux:
diff --git a/packages/flutter/lib/src/material/switch.dart b/packages/flutter/lib/src/material/switch.dart
index d89f4c23ad8..ab96ec9c5c3 100644
--- a/packages/flutter/lib/src/material/switch.dart
+++ b/packages/flutter/lib/src/material/switch.dart
@@ -624,6 +624,7 @@ class Switch extends StatelessWidget {
       case _SwitchType.adaptive:
         switch (Theme.of(context).platform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.linux:
           case TargetPlatform.windows:
@@ -759,6 +760,7 @@ class _MaterialSwitchState extends State<_MaterialSwitch>
           case _SwitchType.adaptive:
             switch (Theme.of(context).platform) {
               case TargetPlatform.android:
+        case TargetPlatform.ohos:
               case TargetPlatform.fuchsia:
               case TargetPlatform.linux:
               case TargetPlatform.windows:
@@ -834,6 +836,7 @@ class _MaterialSwitchState extends State<_MaterialSwitch>
       case _SwitchType.adaptive:
         switch (Theme.of(context).platform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.linux:
           case TargetPlatform.windows:
@@ -935,6 +938,7 @@ class _MaterialSwitchState extends State<_MaterialSwitch>
         switchTheme = switchAdaptation.adapt(theme, switchTheme);
         switch (theme.platform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.linux:
           case TargetPlatform.windows:
@@ -1899,6 +1903,7 @@ class _SwitchThemeAdaptation extends Adaptation<SwitchThemeData> {
   SwitchThemeData adapt(ThemeData theme, SwitchThemeData defaultValue) {
     switch (theme.platform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/material/text_field.dart b/packages/flutter/lib/src/material/text_field.dart
index 7917dbe74e4..491b0988c5a 100644
--- a/packages/flutter/lib/src/material/text_field.dart
+++ b/packages/flutter/lib/src/material/text_field.dart
@@ -85,6 +85,7 @@ class _TextFieldSelectionGestureDetectorBuilder extends TextSelectionGestureDete
         case TargetPlatform.macOS:
           break;
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.linux:
         case TargetPlatform.windows:
@@ -966,6 +967,7 @@ class TextField extends StatefulWidget {
           editableTextState: editableTextState,
         );
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
@@ -1601,6 +1603,7 @@ class _TextFieldState extends State<TextField>
           widget.spellCheckConfiguration,
         );
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
@@ -1661,6 +1664,7 @@ class _TextFieldState extends State<TextField>
         };
 
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.ohos:
         forcePressEnabled = false;
diff --git a/packages/flutter/lib/src/material/theme_data.dart b/packages/flutter/lib/src/material/theme_data.dart
index af035b5e98b..3bc58488468 100644
--- a/packages/flutter/lib/src/material/theme_data.dart
+++ b/packages/flutter/lib/src/material/theme_data.dart
@@ -399,6 +399,7 @@ class ThemeData with Diagnosticable {
     platform ??= defaultTargetPlatform;
     switch (platform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.iOS:
       case TargetPlatform.ohos:
@@ -3252,7 +3253,7 @@ class VisualDensity with Diagnosticable {
   ///   adaptive based on [defaultTargetPlatform].
   static VisualDensity defaultDensityForPlatform(TargetPlatform platform) {
     return switch (platform) {
-      TargetPlatform.android || TargetPlatform.iOS || TargetPlatform.fuchsia || TargetPlatform.ohos => standard,
+      TargetPlatform.android || TargetPlatform.ohos || TargetPlatform.iOS || TargetPlatform.fuchsia || TargetPlatform.ohos => standard,
       TargetPlatform.linux || TargetPlatform.macOS || TargetPlatform.windows => compact,
     };
   }
diff --git a/packages/flutter/lib/src/material/time_picker.dart b/packages/flutter/lib/src/material/time_picker.dart
index 2aa52c15e12..94c61a4bf23 100644
--- a/packages/flutter/lib/src/material/time_picker.dart
+++ b/packages/flutter/lib/src/material/time_picker.dart
@@ -2921,6 +2921,7 @@ class _TimePickerState extends State<_TimePicker> with RestorationMixin {
   void _vibrate() {
     switch (Theme.of(context).platform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/material/tooltip.dart b/packages/flutter/lib/src/material/tooltip.dart
index c1527bc2b30..d5e830d4983 100644
--- a/packages/flutter/lib/src/material/tooltip.dart
+++ b/packages/flutter/lib/src/material/tooltip.dart
@@ -425,7 +425,7 @@ class TooltipState extends State<Tooltip> with SingleTickerProviderStateMixin {
   double _getDefaultTooltipHeight() {
     return switch (Theme.of(context).platform) {
       TargetPlatform.macOS || TargetPlatform.linux || TargetPlatform.windows => 24.0,
-      TargetPlatform.android ||
+      TargetPlatform.android || TargetPlatform.ohos ||
       TargetPlatform.fuchsia ||
       TargetPlatform.iOS ||
       TargetPlatform.ohos => 32.0,
@@ -437,7 +437,7 @@ class TooltipState extends State<Tooltip> with SingleTickerProviderStateMixin {
       TargetPlatform.macOS ||
       TargetPlatform.linux ||
       TargetPlatform.windows => const EdgeInsets.symmetric(horizontal: 8.0, vertical: 4.0),
-      TargetPlatform.android ||
+      TargetPlatform.android || TargetPlatform.ohos ||
       TargetPlatform.fuchsia ||
       TargetPlatform.iOS ||
       TargetPlatform.ohos => const EdgeInsets.symmetric(horizontal: 16.0, vertical: 4.0),
@@ -447,7 +447,7 @@ class TooltipState extends State<Tooltip> with SingleTickerProviderStateMixin {
   static double _getDefaultFontSize(TargetPlatform platform) {
     return switch (platform) {
       TargetPlatform.macOS || TargetPlatform.linux || TargetPlatform.windows => 12.0,
-      TargetPlatform.android ||
+      TargetPlatform.android || TargetPlatform.ohos ||
       TargetPlatform.fuchsia ||
       TargetPlatform.iOS ||
       TargetPlatform.ohos => 14.0,
diff --git a/packages/flutter/lib/src/material/typography.dart b/packages/flutter/lib/src/material/typography.dart
index 8455df6f270..f8afaf9ccc2 100644
--- a/packages/flutter/lib/src/material/typography.dart
+++ b/packages/flutter/lib/src/material/typography.dart
@@ -217,7 +217,7 @@ class Typography with Diagnosticable {
     assert(platform != null || (black != null && white != null));
     final (TextTheme blackResolved, TextTheme whiteResolved) = switch (platform) {
       TargetPlatform.iOS => (black ?? blackCupertino, white ?? whiteCupertino),
-      TargetPlatform.android ||
+      TargetPlatform.android || TargetPlatform.ohos ||
       TargetPlatform.fuchsia => (black ?? blackMountainView, white ?? whiteMountainView),
       TargetPlatform.windows => (black ?? blackRedmond, white ?? whiteRedmond),
       TargetPlatform.macOS => (black ?? blackRedwoodCity, white ?? whiteRedwoodCity),
diff --git a/packages/flutter/lib/src/rendering/editable.dart b/packages/flutter/lib/src/rendering/editable.dart
index e8e1a0e3968..279140e3e3d 100644
--- a/packages/flutter/lib/src/rendering/editable.dart
+++ b/packages/flutter/lib/src/rendering/editable.dart
@@ -1853,6 +1853,7 @@ class RenderEditable extends RenderBox
           caretRect.height,
         );
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
@@ -2261,6 +2262,7 @@ class RenderEditable extends RenderBox
           }
           return TextSelection(baseOffset: previousWord.start, extentOffset: position.offset);
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
           if (readOnly) {
             if (previousWord == null) {
               return TextSelection(baseOffset: position.offset, extentOffset: position.offset + 1);
@@ -2338,6 +2340,7 @@ class RenderEditable extends RenderBox
       case TargetPlatform.macOS:
         _caretPrototype = Rect.fromLTWH(0.0, 0.0, cursorWidth, cursorHeight + 2);
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/rendering/view.dart b/packages/flutter/lib/src/rendering/view.dart
index 3e31ef42b2f..8cf33f403a7 100644
--- a/packages/flutter/lib/src/rendering/view.dart
+++ b/packages/flutter/lib/src/rendering/view.dart
@@ -431,6 +431,7 @@ class RenderView extends RenderObject with RenderObjectWithChildMixin<RenderBox>
     SystemUiOverlayStyle? lowerOverlayStyle;
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
         lowerOverlayStyle = layer!.find<SystemUiOverlayStyle>(bottom);
       case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
diff --git a/packages/flutter/lib/src/services/system_navigator.dart b/packages/flutter/lib/src/services/system_navigator.dart
index b5246f2ae86..f9d55c688dc 100644
--- a/packages/flutter/lib/src/services/system_navigator.dart
+++ b/packages/flutter/lib/src/services/system_navigator.dart
@@ -37,6 +37,7 @@ abstract final class SystemNavigator {
       case TargetPlatform.ohos:
         return;
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
         return SystemChannels.platform.invokeMethod<void>(
           'SystemNavigator.setFrameworkHandlesBack',
           frameworkHandlesBack,
diff --git a/packages/flutter/lib/src/services/text_formatter.dart b/packages/flutter/lib/src/services/text_formatter.dart
index a2f729a745a..9c458fe243a 100644
--- a/packages/flutter/lib/src/services/text_formatter.dart
+++ b/packages/flutter/lib/src/services/text_formatter.dart
@@ -520,6 +520,7 @@ class LengthLimitingTextInputFormatter extends TextInputFormatter {
     } else {
       switch (platform ?? defaultTargetPlatform) {
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.windows:
           return MaxLengthEnforcement.enforced;
         case TargetPlatform.iOS:
diff --git a/packages/flutter/lib/src/widgets/autocomplete.dart b/packages/flutter/lib/src/widgets/autocomplete.dart
index 15a0e9d6f2e..bbfbe738fd2 100644
--- a/packages/flutter/lib/src/widgets/autocomplete.dart
+++ b/packages/flutter/lib/src/widgets/autocomplete.dart
@@ -412,7 +412,7 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
     ...switch (defaultTargetPlatform) {
       TargetPlatform.iOS => _appleShortcuts,
       TargetPlatform.macOS => _appleShortcuts,
-      TargetPlatform.android => _nonAppleShortcuts,
+      TargetPlatform.android || TargetPlatform.ohos => _nonAppleShortcuts,
       TargetPlatform.linux => _nonAppleShortcuts,
       TargetPlatform.windows => _nonAppleShortcuts,
       TargetPlatform.fuchsia => _nonAppleShortcuts,
diff --git a/packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart b/packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart
index 5d43394b257..fb6f8a66658 100644
--- a/packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart
+++ b/packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart
@@ -1012,7 +1012,7 @@ class DefaultTextEditingShortcuts extends StatelessWidget {
 
   static Map<ShortcutActivator, Intent> get _shortcuts {
     return switch (defaultTargetPlatform) {
-      TargetPlatform.android => _androidShortcuts,
+      TargetPlatform.android || TargetPlatform.ohos => _androidShortcuts,
       TargetPlatform.fuchsia => _fuchsiaShortcuts,
       TargetPlatform.iOS => _iOSShortcuts,
       TargetPlatform.linux => _linuxShortcuts,
@@ -1032,6 +1032,7 @@ class DefaultTextEditingShortcuts extends StatelessWidget {
               activator as SingleActivator: const DoNothingAndStopPropagationTextIntent(),
           };
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.windows:
         case TargetPlatform.iOS:
@@ -1042,6 +1043,7 @@ class DefaultTextEditingShortcuts extends StatelessWidget {
     }
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/widgets/editable_text.dart b/packages/flutter/lib/src/widgets/editable_text.dart
index 9a0aaf203cf..148f8b9956e 100644
--- a/packages/flutter/lib/src/widgets/editable_text.dart
+++ b/packages/flutter/lib/src/widgets/editable_text.dart
@@ -2113,7 +2113,7 @@ class EditableText extends StatefulWidget {
       return true;
     }
     return switch (defaultTargetPlatform) {
-      TargetPlatform.android => false,
+      TargetPlatform.android || TargetPlatform.ohos => false,
       TargetPlatform.ohos => false,
       TargetPlatform.iOS => false,
       TargetPlatform.fuchsia => false,
@@ -2286,6 +2286,7 @@ class EditableText extends StatefulWidget {
             return keyboardType;
           }
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.linux:
         case TargetPlatform.windows:
@@ -2685,6 +2686,7 @@ class EditableTextState extends State<EditableText>
       case TargetPlatform.iOS:
         return textEditingValue.text.isNotEmpty && textEditingValue.selection.isCollapsed;
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
@@ -2720,6 +2722,7 @@ class EditableTextState extends State<EditableText>
   bool get shareEnabled {
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.iOS:
         return !widget.obscureText &&
             !textEditingValue.selection.isCollapsed &&
@@ -2784,6 +2787,7 @@ class EditableTextState extends State<EditableText>
           break;
         case TargetPlatform.ohos:
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
           // Collapse the selection and hide the toolbar and handles.
           userUpdateTextEditingValue(
@@ -2915,6 +2919,7 @@ class EditableTextState extends State<EditableText>
     if (cause == SelectionChangedCause.toolbar) {
       switch (defaultTargetPlatform) {
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.iOS:
         case TargetPlatform.fuchsia:
         case TargetPlatform.ohos:
@@ -2926,6 +2931,7 @@ class EditableTextState extends State<EditableText>
       }
       switch (defaultTargetPlatform) {
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.linux:
         case TargetPlatform.windows:
@@ -4202,7 +4208,7 @@ class EditableTextState extends State<EditableText>
   }
 
   final bool _platformSupportsFadeOnScroll = switch (defaultTargetPlatform) {
-    TargetPlatform.android || TargetPlatform.iOS => true,
+    TargetPlatform.android || TargetPlatform.ohos || TargetPlatform.iOS => true,
     TargetPlatform.fuchsia ||
     TargetPlatform.linux ||
     TargetPlatform.macOS ||
@@ -7092,6 +7098,7 @@ class _EditableTextTapOutsideAction extends ContextAction<EditableTextTapOutside
     // The focus dropping behavior is only present on desktop platforms.
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.iOS:
       case TargetPlatform.fuchsia:
       case TargetPlatform.ohos:
diff --git a/packages/flutter/lib/src/widgets/feedback.dart b/packages/flutter/lib/src/widgets/feedback.dart
index 3ef0bca9942..e59fe463473 100644
--- a/packages/flutter/lib/src/widgets/feedback.dart
+++ b/packages/flutter/lib/src/widgets/feedback.dart
@@ -92,6 +92,7 @@ abstract final class Feedback {
     context.findRenderObject()!.sendSemanticsEvent(const TapSemanticEvent());
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
         return SystemSound.play(SystemSoundType.click);
       case TargetPlatform.iOS:
@@ -138,6 +139,7 @@ abstract final class Feedback {
     context.findRenderObject()!.sendSemanticsEvent(const LongPressSemanticsEvent());
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
         return HapticFeedback.vibrate();
       case TargetPlatform.iOS:
diff --git a/packages/flutter/lib/src/widgets/focus_manager.dart b/packages/flutter/lib/src/widgets/focus_manager.dart
index 7af9af5a184..764c4ac300a 100644
--- a/packages/flutter/lib/src/widgets/focus_manager.dart
+++ b/packages/flutter/lib/src/widgets/focus_manager.dart
@@ -1679,7 +1679,7 @@ class FocusManager with DiagnosticableTreeMixin, ChangeNotifier {
   bool get _respondToLifecycleChange =>
       kIsWeb ||
       switch (defaultTargetPlatform) {
-        TargetPlatform.android || TargetPlatform.iOS || TargetPlatform.ohos => false,
+        TargetPlatform.android || TargetPlatform.ohos || TargetPlatform.iOS || TargetPlatform.ohos => false,
         TargetPlatform.fuchsia || TargetPlatform.linux => true,
         TargetPlatform.windows || TargetPlatform.macOS => true,
       };
@@ -2374,6 +2374,7 @@ class _HighlightModeManager {
     // known correct value as soon as any pointer/keyboard events are received.
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.iOS:
       case TargetPlatform.ohos:
diff --git a/packages/flutter/lib/src/widgets/modal_barrier.dart b/packages/flutter/lib/src/widgets/modal_barrier.dart
index c21f1e6b845..d463913b716 100644
--- a/packages/flutter/lib/src/widgets/modal_barrier.dart
+++ b/packages/flutter/lib/src/widgets/modal_barrier.dart
@@ -215,6 +215,7 @@ class ModalBarrier extends StatelessWidget {
       case TargetPlatform.ohos:
         platformSupportsDismissingBarrier = false;
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.iOS:
       case TargetPlatform.macOS:
         platformSupportsDismissingBarrier = true;
diff --git a/packages/flutter/lib/src/widgets/navigator.dart b/packages/flutter/lib/src/widgets/navigator.dart
index c2e583e849c..07935d2bb68 100644
--- a/packages/flutter/lib/src/widgets/navigator.dart
+++ b/packages/flutter/lib/src/widgets/navigator.dart
@@ -3308,6 +3308,7 @@ class _RouteEntry extends RouteTransitionRecord {
           .then((dynamic result) async {
             switch (defaultTargetPlatform) {
               case TargetPlatform.android:
+        case TargetPlatform.ohos:
                 // In the Android platform, we have to wait for the system refocus to complete before
                 // sending the refocus message. Otherwise, the refocus message will be ignored.
                 // TODO(hangyujin): update this logic if Android provide a better way to do so.
diff --git a/packages/flutter/lib/src/widgets/platform_menu_bar.dart b/packages/flutter/lib/src/widgets/platform_menu_bar.dart
index 3c599114527..bd8bbc98385 100644
--- a/packages/flutter/lib/src/widgets/platform_menu_bar.dart
+++ b/packages/flutter/lib/src/widgets/platform_menu_bar.dart
@@ -885,6 +885,7 @@ class PlatformProvidedMenuItem extends PlatformMenuItem {
   static bool hasMenu(PlatformProvidedMenuItemType menu) {
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.iOS:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
diff --git a/packages/flutter/lib/src/widgets/scroll_configuration.dart b/packages/flutter/lib/src/widgets/scroll_configuration.dart
index 233911e92a9..ab0357e1392 100644
--- a/packages/flutter/lib/src/widgets/scroll_configuration.dart
+++ b/packages/flutter/lib/src/widgets/scroll_configuration.dart
@@ -132,6 +132,7 @@ class ScrollBehavior {
       case TargetPlatform.linux:
       case TargetPlatform.windows:
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.ohos:
         return MultitouchDragStrategy.latestPointer;
@@ -168,6 +169,7 @@ class ScrollBehavior {
         assert(details.controller != null);
         return RawScrollbar(controller: details.controller, child: child);
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.iOS:
       case TargetPlatform.ohos:
@@ -188,6 +190,7 @@ class ScrollBehavior {
       case TargetPlatform.ohos:
         return child;
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
         return GlowingOverscrollIndicator(
           axisDirection: details.direction,
@@ -253,6 +256,7 @@ class ScrollBehavior {
       case TargetPlatform.macOS:
         return _bouncingDesktopPhysics;
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
diff --git a/packages/flutter/lib/src/widgets/scrollbar.dart b/packages/flutter/lib/src/widgets/scrollbar.dart
index 12d3aa444df..c85e5abdf94 100644
--- a/packages/flutter/lib/src/widgets/scrollbar.dart
+++ b/packages/flutter/lib/src/widgets/scrollbar.dart
@@ -1890,7 +1890,7 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
     // dragging on the trackpad or with a stylus.
     final TargetPlatform platform = ScrollConfiguration.of(context).getPlatform(context);
     final Velocity adjustedVelocity = switch (platform) {
-      TargetPlatform.iOS || TargetPlatform.android => -velocity,
+      TargetPlatform.iOS || TargetPlatform.android || TargetPlatform.ohos => -velocity,
       _ => Velocity.zero,
     };
     final renderBox = _scrollbarPainterKey.currentContext!.findRenderObject()! as RenderBox;
diff --git a/packages/flutter/lib/src/widgets/selectable_region.dart b/packages/flutter/lib/src/widgets/selectable_region.dart
index 22f83a10ac3..6754ae694e3 100644
--- a/packages/flutter/lib/src/widgets/selectable_region.dart
+++ b/packages/flutter/lib/src/widgets/selectable_region.dart
@@ -307,7 +307,7 @@ class SelectableRegion extends StatefulWidget {
     final bool platformCanShare =
         !kIsWeb &&
         switch (defaultTargetPlatform) {
-          TargetPlatform.android => selectionGeometry.status == SelectionStatus.uncollapsed,
+          TargetPlatform.android || TargetPlatform.ohos => selectionGeometry.status == SelectionStatus.uncollapsed,
           TargetPlatform.macOS ||
           TargetPlatform.fuchsia ||
           TargetPlatform.linux ||
@@ -475,6 +475,7 @@ class SelectableRegionState extends State<SelectableRegion>
     super.didChangeDependencies();
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.iOS:
         break;
       case TargetPlatform.ohos:
@@ -605,6 +606,7 @@ class SelectableRegionState extends State<SelectableRegion>
     var maxConsecutiveTap = 3;
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
         if (_lastPointerDeviceKind != null && _lastPointerDeviceKind != PointerDeviceKind.mouse) {
           // When the pointer device kind is not precise like a mouse, native
@@ -739,6 +741,7 @@ class SelectableRegionState extends State<SelectableRegion>
         _focusNode.requestFocus();
         switch (defaultTargetPlatform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.iOS:
           case TargetPlatform.ohos:
@@ -778,6 +781,7 @@ class SelectableRegionState extends State<SelectableRegion>
               _showHandles();
             }
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.macOS:
           case TargetPlatform.linux:
@@ -789,6 +793,7 @@ class SelectableRegionState extends State<SelectableRegion>
       case 3:
         switch (defaultTargetPlatform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.iOS:
           case TargetPlatform.ohos:
@@ -833,6 +838,7 @@ class SelectableRegionState extends State<SelectableRegion>
       case 2:
         switch (defaultTargetPlatform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.ohos:
             // Double tap + drag is only supported on Android when using a precise
@@ -877,6 +883,7 @@ class SelectableRegionState extends State<SelectableRegion>
       case 3:
         switch (defaultTargetPlatform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.iOS:
           case TargetPlatform.ohos:
@@ -914,6 +921,7 @@ class SelectableRegionState extends State<SelectableRegion>
     final bool shouldShowSelectionOverlayOnMobile = !isPointerPrecise;
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.ohos:
         if (shouldShowSelectionOverlayOnMobile) {
@@ -952,6 +960,7 @@ class SelectableRegionState extends State<SelectableRegion>
       case 1:
         switch (defaultTargetPlatform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.iOS:
           case TargetPlatform.ohos:
@@ -967,6 +976,7 @@ class SelectableRegionState extends State<SelectableRegion>
         final bool isPointerPrecise = _isPrecisePointerDevice(details.kind);
         switch (defaultTargetPlatform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.ohos:
             if (!isPointerPrecise) {
@@ -1056,6 +1066,7 @@ class SelectableRegionState extends State<SelectableRegion>
     _focusNode.requestFocus();
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.windows:
       case TargetPlatform.ohos:
@@ -1715,6 +1726,7 @@ class SelectableRegionState extends State<SelectableRegion>
         // On Android copy should clear the selection.
         switch (defaultTargetPlatform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.ohos:
             clearSelection();
@@ -1730,6 +1742,7 @@ class SelectableRegionState extends State<SelectableRegion>
       onSelectAll: () {
         switch (defaultTargetPlatform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.iOS:
           case TargetPlatform.fuchsia:
           case TargetPlatform.ohos:
@@ -1747,6 +1760,7 @@ class SelectableRegionState extends State<SelectableRegion>
         // On Android, share should clear the selection.
         switch (defaultTargetPlatform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.ohos:
             clearSelection();
diff --git a/packages/flutter/lib/src/widgets/text_selection.dart b/packages/flutter/lib/src/widgets/text_selection.dart
index d10bf8d1056..61171cc1495 100644
--- a/packages/flutter/lib/src/widgets/text_selection.dart
+++ b/packages/flutter/lib/src/widgets/text_selection.dart
@@ -824,6 +824,7 @@ class TextSelectionOverlay {
           extentOffset: position.offset,
         );
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
@@ -956,6 +957,7 @@ class TextSelectionOverlay {
           extentOffset: position.offset,
         );
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
@@ -1457,6 +1459,7 @@ class SelectionOverlay {
       if (_isDraggingEndHandle || _isDraggingStartHandle) {
         switch (defaultTargetPlatform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
             HapticFeedback.selectionClick();
           case TargetPlatform.fuchsia:
           case TargetPlatform.iOS:
@@ -2217,6 +2220,7 @@ class TextSelectionGestureDetectorBuilder {
   void _showMagnifierIfSupportedByPlatform(Offset positionToShow) {
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.iOS:
         editableText.showMagnifier(positionToShow);
       case TargetPlatform.fuchsia:
@@ -2235,6 +2239,7 @@ class TextSelectionGestureDetectorBuilder {
 
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.iOS:
         editableText.hideMagnifier();
       case TargetPlatform.fuchsia:
@@ -2470,6 +2475,7 @@ class TextSelectionGestureDetectorBuilder {
         _isShiftPressed && renderEditable.selection?.baseOffset != null;
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
         if (editableText.widget.stylusHandwritingEnabled) {
           final bool stylusEnabled = switch (kind) {
             PointerDeviceKind.stylus ||
@@ -2614,6 +2620,7 @@ class TextSelectionGestureDetectorBuilder {
       // On desktop platforms the selection is set on tap down.
       case TargetPlatform.ohos:
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
         editableText.hideToolbar(false);
         if (isShiftPressedValid) {
           _extendSelection(details.globalPosition, SelectionChangedCause.tap);
@@ -2770,6 +2777,7 @@ class TextSelectionGestureDetectorBuilder {
           editableText.updateFloatingCursor(cursorPoint);
         }
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
@@ -2842,6 +2850,7 @@ class TextSelectionGestureDetectorBuilder {
           editableText.updateFloatingCursor(cursorPoint);
         }
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
@@ -2911,6 +2920,7 @@ class TextSelectionGestureDetectorBuilder {
           editableText.showToolbar();
         }
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.ohos:
@@ -3078,6 +3088,7 @@ class TextSelectionGestureDetectorBuilder {
     } else {
       switch (defaultTargetPlatform) {
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.iOS:
         case TargetPlatform.macOS:
@@ -3129,6 +3140,7 @@ class TextSelectionGestureDetectorBuilder {
         case TargetPlatform.macOS:
           _expandSelection(details.globalPosition, SelectionChangedCause.drag);
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.linux:
         case TargetPlatform.windows:
@@ -3152,6 +3164,7 @@ class TextSelectionGestureDetectorBuilder {
             case null:
           }
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.ohos:
           switch (details.kind) {
@@ -3246,6 +3259,7 @@ class TextSelectionGestureDetectorBuilder {
           3) {
         switch (defaultTargetPlatform) {
           case TargetPlatform.android:
+        case TargetPlatform.ohos:
           case TargetPlatform.fuchsia:
           case TargetPlatform.iOS:
           case TargetPlatform.ohos:
@@ -3306,6 +3320,7 @@ class TextSelectionGestureDetectorBuilder {
           }
           return;
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.ohos:
           // With a precise pointer device, such as a mouse, trackpad, or stylus,
@@ -3602,6 +3617,7 @@ class _TextSelectionGestureDetectorState extends State<TextSelectionGestureDetec
   static int _getEffectiveConsecutiveTapCount(int rawCount) {
     switch (defaultTargetPlatform) {
       case TargetPlatform.android:
+        case TargetPlatform.ohos:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.ohos:
@@ -3741,6 +3757,7 @@ class _TextSelectionGestureDetectorState extends State<TextSelectionGestureDetec
         widget.onDragSelectionEnd != null) {
       switch (defaultTargetPlatform) {
         case TargetPlatform.android:
+        case TargetPlatform.ohos:
         case TargetPlatform.fuchsia:
         case TargetPlatform.iOS:
         case TargetPlatform.ohos:
